### PR TITLE
ME-15 Fix ARSerifNavigationBar element positions

### PR DESCRIPTION
**Pull request by ds300**
https://artsyproduct.atlassian.net/browse/ME-15

## Before

bad-nav-bar

## After

good-nav-bar

### CODE REVIEW

File: Artsy/View_Controllers/ARSerifNavigationViewController.m [hunk 1]
```diff
@@ -274,21 +284,24 @@ - (void)verticallyCenterView:(id)viewOrArray
...
-    CGFloat barMidpoint = roundf(self.frame.size.height / 2);
-    CGFloat viewMidpoint = roundf(viewToCenter.frame.size.height / 2);
+    CGFloat yOffset = 0;
+    if ([viewToCenter isDescendantOfView:self]) {
+        UIView *child = viewToCenter.superview;
+        int safety = 0;
+        while (safety++ < 10 && child != self) {
+            yOffset += child.frame.origin.y;
+            child = child.superview;
+        }
```
Comment (ashfurrow) [inline Artsy/View_Controllers/ARSerifNavigationViewController.m]: It seems like what this loop is trying to find the vertical difference in frames between `self` and `viewToCenter`. We can use a few methods in `UIView` to accomplish this as well:

```objc
// Grab the bounds, which is viewToCenter's CGRect within its own coordinate space, and covert to self's space
CGRect viewToCenterFrameInSelf = [viewToCenter convertRect:viewToCenter.bounds toCoordinateSpace:self];
CGFloat yOffset = viewToCenterFrameInSelf.origin.y;
```

I _think_ that should do it.

**Comment by alloy** [comment]
> It's not an ideal workflow, maybe there's a reason we haven't documented it yet 🙈 I'm very open to suggestions here. /cc @alloy

I don’t know of another solution. I suspect not much time was put into it because the need for it doesn’t arise often.

File: Artsy/View_Controllers/ARSerifNavigationViewController.m (hunk 1 above)
Comment (ds300) [reply Artsy/View_Controllers/ARSerifNavigationViewController.m]: Hmm this is not working. Looks like this:

image

Also tried converting the bounds to window space before calculating the offset, same result. 🤔 

Any other approaches you think might work? I'm not crazy about this while look either.

Comment (ds300) [reply Artsy/View_Controllers/ARSerifNavigationViewController.m]: could it be that the height of the text label is scaled down somewhere and needs to be converted too?

Comment (ds300) [reply Artsy/View_Controllers/ARSerifNavigationViewController.m]: it could not

Comment (ashfurrow) [reply Artsy/View_Controllers/ARSerifNavigationViewController.m]: Mmm, [`locationInView:`](https://developer.apple.com/documentation/uikit/uitouch/1618116-locationinview?language=objc) might be helpful here? Agreed the loop isn't ideal, but if it works (and has `safety`!) then I'm okay shipping it. Maybe add a comment explaining what we tried instead, for posterity?

Comment (ds300) [reply Artsy/View_Controllers/ARSerifNavigationViewController.m]: Unless I missed something, locationInView is a method of UITouch, which I don't have an instance of?

I think let's stick with the while loop. I added a comment, thanks for the suggestion :+1:

**Comment by ds300** [comment]
This should be 🍏

**Pull request merged**